docs/use_bls_wallet_dapp.md

```diff
@@ -0,0 +1,109 @@
+
+## Download, Install, & Setup Quill
+
+[Quill](../extension/) is a protoype browser extension wallet which intergrates [bls-wallet-clients](../contracts/clients/) to communicate with the BLS Wallet smart contracts & transaction aggregator. It supports most of the functionality in [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193).
+
+Currently, we have the contracts deployed to the networks/chains listed [here](https://github.com/web3well/bls-wallet/tree/main/contracts/networks). If your desired networks isn't there, you can use the [Remote Development](./remote_development.md) contract deployment intrsuctions or request a network deploy by [opening an issue](https://github.com/web3well/bls-wallet/issues/new) or [starting a discussion](https://github.com/web3well/bls-wallet/discussions/new).
+
+Below are the instructions for 2 ways you can add Quill to your browser.
+
+### Prebuilt
+
+Go to the [releases page](https://github.com/web3well/bls-wallet/releases) and scroll down to the latest release. In the `Assets` section, download the extension for either Chrome, Firefox, or Opera. To install, simply drag and drop the file into your browser on the extensions page or follow instructions for installing extensions from a file for your browser.
+
+### From Repo
+
+Follow the instructions in either [Local Development](./local_development.md) or [Remote Development](./remote_development.md) to setup this repo and install Quill.
+
+### Setup Quill
+
+Folow the setup instructions when ou first open Quill, and then create a wallet if there is not one there already.
+
+## Connect Your dApp to Quill
+
+Next, connect your dApp to Quill just like you would any other extension wallet.
+
+```typescript
+const provider = providers.Web3Provider(window.ethereum);
+
+await window.ethereum.request({ method: "eth_accounts" });
+```
+
+## Send Multi-Action Transaction
+
+Finally, you can populate & send your multi-action transaction. In the following example, we will do an approve & swap with a DEX (decentralized exchange) in one transaction.
+
+### Check that window.ethereum Supports Multi-Action Transactions
+
+Since any browser extension wallet could be used, make sure that it is Quill before allowing a multi-action transaction.
+
+```typescript
+const areMultiActionTransactionSupported = "isQuill" in window.ethereum && window.ethereum.isQuill;
+// Branch here depending on the result
+if (areMultiActionTransactionSupported) {
+  ...
+}
+```
+
+### Populate the Transactions (Actions)
+
+First, we will populate the transactions (actions) we want to send.
+
+```typescript
+// Connect to the contracts and get the signer.
+const erc20Contract = new ethers.Contract(erc20Address, erc20Abi, provider);
+const dexContract = new ethers.Contract(dexAddress, dex20Abi, provider);
+
+const signer = provider.getSigner();
+
+// Populate the token approval transaction.
+const approveTransaction = await erc20Contract
+  .connect(signer)
```

Comment:
I don't think `.connect(signer)` is necessary here (the contract was already initialized with this provider that contains the signer). Could you double-check this?

Reply:
I was mainly going off @kautukkundan 's examples, but yeah I can make sure that step is needed.

Reply:
Always a good idea to include the signer in tx request as there may be a requirement/dependency for `msg.sender` in the function itself. The provider itself may or may not include the correct signer. I do this for `callstatic()` always and think that it should be included here as well

Reply:
- If you pass a provider into a contract and attempt to change state (write), it will throw an error.
- If you pass the signer into the contract, you can now change state.

Given that the [ether.js example](https://docs.ethers.io/v5/getting-started/#getting-started--writing) uses `connect` and it better separates read/write concerns, I'm going to leave it in with a note above the contract instantiation line about how you can also pass the signer in.

Reply:
Updated